native/markdown/rules.react.js

```diff
@@ -49,61 +57,45 @@ export default function rules(styles: StyleSheetOf<MarkdownStyles>) {
       ...SimpleMarkdown.defaultRules.paragraph,
       // simple-markdown collapses multiple newlines into one, but we want to
       // preserve the newlines
-      match: SimpleMarkdown.blockRegex(/^([^\n]*)(?:\n|$)/),
+      match: SimpleMarkdown.blockRegex(/^((?:[^\n]*)(?:\n|$))/),
       // eslint-disable-next-line react/display-name
       react: (
         node: SimpleMarkdown.SingleASTNode,
         output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
         state: SimpleMarkdown.State,
-      ) => {
-        let innerNode = output(node.content, state);
-        if (Array.isArray(innerNode) && innerNode.length === 0) {
-          // React Native renders multiple empty lines within their own Text
-          // differently than it renders a single Text that contains those empty
-          // lines within it. We render this space to make sure it renders as if
-          // it were a single Text node
-          innerNode = ' ';
-        }
-        return (
-          <Text key={state.key} style={styles.paragraph}>
-            {innerNode}
-          </Text>
-        );
-      },
+      ) => (
+        <React.Fragment key={state.key}>
+          {output(node.content, state)}
+        </React.Fragment>
+      ),
     },
     // This is the leaf node in the AST returned by the parse phase. We handle
     // rendering emoji as a different size here
     text: {
       ...SimpleMarkdown.defaultRules.text,
-      react(
+      // eslint-disable-next-line react/display-name
+      react: (
         node: SimpleMarkdown.SingleASTNode,
         output: SimpleMarkdown.Output<string>,
         state: SimpleMarkdown.State,
-      ) {
-        const { textStyle } = state;
-        invariant(
-          textStyle && typeof textStyle === 'object',
-          `state passed to Markdown output should have object textStyle`,
-        );
-
-        const style = [textStyle];
-        if (state.linkPresent) {
-          style.push(styles.link);
-        } else if (state.emojiOnly) {
-          const { fontSize } = textStyle;
-          invariant(
-            fontSize,
-            `textStyle should have fontSize if using emojiOnly`,
-          );
-          style.push({ fontSize: fontSize * 2 });
-        }
-
-        return (
-          <Text key={state.key} style={style}>
-            {node.content}
-          </Text>
-        );
-      },
+      ) => <React.Fragment key={state.key}>{node.content}</React.Fragment>,
```

Comment:
Even cleaner! The `Markdown` component handles `emojiOnly` now, and the `link` rule handles itself